src/battle.ts

```diff
@@ -2941,22 +2953,14 @@ class Battle {
 		let siden = -1;
 		let slot = -1; // if there is an explicit slot for this pokemon
 		let slotChart: {[k: string]: number} = {a: 0, b: 1, c: 2, d: 3, e: 4, f: 5};
-		if (name.substr(0, 4) === 'p2: ' || name === 'p2') {
-			siden = this.p2.n;
-			name = name.substr(4);
-		} else if (name.substr(0, 4) === 'p1: ' || name === 'p1') {
-			siden = this.p1.n;
-			name = name.substr(4);
-		} else if (name.substr(0, 2) === 'p2' && name.substr(3, 2) === ': ') {
-			slot = slotChart[name.substr(2, 1)];
-			siden = this.p2.n;
-			name = name.substr(5);
-			pokemonid = 'p2: ' + name;
-		} else if (name.substr(0, 2) === 'p1' && name.substr(3, 2) === ': ') {
-			slot = slotChart[name.substr(2, 1)];
-			siden = this.p1.n;
-			name = name.substr(5);
-			pokemonid = 'p1: ' + name;
+		if (/^p[1-9]($|: )/.test(name)) {
+			siden = parseInt(name.charAt(1)) - 1;
+			name = name.slice(4);
+		} else if (/^p[1-9][a-f]: /.test(name))
+			siden = parseInt(name.charAt(1)) - 1;
+			slot = slotChart[name.charAt(2));
+			name = name.slice(5);
+			pokemonid = `p${siden + 1}: ${name}`;
 		}
 		return {name, siden, slot, pokemonid};
```

Comment:
All of a sudden this isn't valid syntax anymore and is causing build errors. What gives?